src/consumer/component/consumer-component.js

```diff
@@ -628,16 +628,15 @@ export default class Component {
       impl = path.join(bitDir, bitJson.getImplBasename());
       specs = path.join(bitDir, bitJson.getSpecBasename());
     }
-
+    const entry = fs.existsSync(path.join(process.cwd(), bitJson.distEntry)) ? path.join(process.cwd(), bitJson.distEntry) : process.cwd()
     const files = componentMap.files;
-    const vinylFiles = Object.keys(files).map((file) => vinylFile.readSync(path.join(consumerPath, files[file])));
+    const vinylFiles = Object.keys(files).map(file =>  new Files(vinylFile.readSync(path.join(consumerPath, files[file]),{base:entry})));
     // TODO: Decide about the model represntation
     componentMap.testsFiles.forEach(testFile => {
-      let file = vinylFile.readSync(path.join(consumerPath, testFile))
+      let file = vinylFile.readSync(path.join(consumerPath, testFile), {base:entry})
```

Comment:
Why are you using read sync instead of regular read?
You can run this in parallel to improve performance.
Make sure you run it correctly with promise.all